### DISCUSSION TITLE
Error: Error E138 is reported when exiting nvim

### DISCUSSION THREAD

**Discussion by iamxiaojianzheng** [LazyVim/LazyVim#3669] [Q&A]
E138: All ...\AppData\Local\nvim-data\shada\main.shada.tmp.X files exist, cannot write ShaDa file!

**Comment by iamxiaojianzheng** [comment]
This question comes up a bit frequently for me.

Other friends who use neovim have encountered this kind of problem?

**Reply by dpetka2001** [reply]
If you're on Windows (which I guess you are from the filepath), then I've read multiple issues about users reporting this error with Shada. It seems to be related to Neovim upstream on Windows. Something must be happening when exiting Neovim and can't write correctly the Shada file, thus the `.tmp` get created. Delete all those files in the directory mentioned.

**Reply by iamxiaojianzheng** [reply]
Yes, the solution is to delete the file. 

It's just that this mistake happens a little too often, so I wanted to ask.

I found that someone in neovim's issues raised this issue and it seems to have been resolved. Look forward to the next release of neovim.

https://github.com/neovim/neovim/issues/8587

https://github.com/neovim/neovim/issues/29186

fix pr https://github.com/neovim/neovim/pull/29093

**Comment by PAMinerva** [comment]
I still occasionally get this error message on Windows 11, so I temporarily solved it by appending the following code to lua/config/autocmds.lua:

```
local function CleanShaDaTmp()
  local shada_dir = vim.fn.stdpath("data") .. "\\shada"
  local pattern = shada_dir .. "\\main.shada.tmp.*"

  for _, file in ipairs(vim.fn.glob(pattern, false, true)) do
    os.remove(file)
  end
end

vim.api.nvim_create_autocmd("VimLeavePre", {
  callback = CleanShaDaTmp,
})
```

However, I’d like this to be properly fixed.
